src/journey.js

```diff
@@ -23,6 +26,35 @@ exports.create = (event, context, callback) => {
     return db.simple_create(event,newItem,callback);
 }
 
+exports.compile = (event) => {
+    const reqBody = JSON.parse(event.body);
+
+    // The current DB structure is problematic for this, as a crafted
+    // request could be used to retrieve other records
+    let params = {
+        RequestItems: {
+            [tableName]: {
+                Keys: reqBody.journeys.map(x => {
+                    return {id: x }
+                }),
+                ProjectionExpression: "doc"
+            }
+        }
+    };
+    
+    return db.dynamo.batchGet(params).promise()
+      .then( (data) => {
+        const docs = data.Responses[tableName]
+        const pages = docs.map(x => x.doc.pages)
+        const fields = pages.reduce((a, b) => [...a, ...b])
+
```

Comment:
I think flatMap would do this nicely. Something like
`const fields = docs.flatMap( x => x.doc.pages)`